council/utils/code_parser.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+from more_itertools import first, last
+import re
+from typing import Optional, List, Iterable
+
+
+class CodeBlock:
+    def __init__(self, language: str, code: str):
+        self._language = language if len(language) > 0 else "unknown"
+        self._code = code
```

Comment:
How about:
-  using `None` instead of `unknown`
-  considering the `language` is potentially optional, I'd change `__init__` as `def __init__(self, code: str, language: Optional[str] = None):`